lib/index.js

```diff
@@ -3,32 +3,48 @@
 const Bluebird = require('bluebird');
 const Joi      = require('joi');
 
-const Validator = require('./validator');
-
-const RADIX = 10;
+const ApproximateCountValidator = require('./validators/approximate-count');
+const Counter                   = require('./counter');
+const Key                       = require('./key');
+const TotalCountValidator       = require('./validators/total-count');
 
 exports.register = (server, options, next) => {
 
   server.ext('onPreResponse', (request, reply) => {
 
     const settings = request.route.settings.plugins.totalCount;
-    const credentials = request.auth.credentials;
-    const validation = Joi.validate(request, Validator, { allowUnknown: true });
+    const approximateCountValid = !Joi.validate(request, ApproximateCountValidator, { allowUnknown: true }).error;
+    const totalCountValid = !Joi.validate(request, TotalCountValidator, { allowUnknown: true }).error;
 
-    if (!settings || validation.error) {
+    if (!settings || !settings.model || !approximateCountValid && !totalCountValid) {
```

Comment:
Can we add parenthesis around `!approximateCountValid && !totalCountValid` to make the condition explicit rather than rely on left-to-right parsing? I think it has the right behavior but it can get confusing.